src/radical/entk/appman/appmanager.py

```diff
@@ -842,6 +842,11 @@ def _update_task(self, msg, reply_to, corr_id, mq_channel, method_frame):
                             self._logger.debug('Task %s path set to %s' %
                                                (task.uid, task.path))
 
+                        if completed_task.rts_uid:
```

Comment:
Should we maybe set this for running tasks already?  Otherwise the RTS uid will not be available for tasks which did not complete, for some reason?  Probably not much of a problem though...

Reply:
I think it is set every time the app manager gets an update for a task and not only when they are completed. The method is called `_update_task` and the variable `completed_task`.

I will check and let you know

Reply:
The function that sets information like `path` and `rts_uid` for a task is called `create_task_from_cu` which is called in a unit call back ([here](https://github.com/radical-cybertools/radical.entk/blob/devel/src/radical/entk/execman/rp/task_manager.py#L249)). The callback only creates a task from a unit when the unit has finished executing. Every other task state update the Appmanager receives comes from somewhere else.

> Otherwise the RTS uid will not be available for tasks which did not complete, for some reason?

That is true. I do not think we should care about this case.

Reply:
If that implies structural changes to the code, then it's out of scope for this PR indeed.  Thanks for checking.